main.py

```diff
@@ -141,14 +143,44 @@ def login(driver, username, password):
     email_input = driver.find_element_by_xpath("//input[@name='emailAddress']")
     email_input.clear()
     email_input.send_keys(username)
+    
     password_input = driver.find_element_by_xpath("//input[@name='password']")
     password_input.clear()
     password_input.send_keys(password)
 
     LOGGER.info("Logging in")
     driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
-    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']")
+    
+    sleep_val = 3
+    num_retries_attempted = 0
+    num_retries = 5
+    while True:
+        try:
+            time.sleep(sleep_val)
+            xpath = "/html/body/div[2]/div[3]/div[3]/div/div[2]/input"
+            wait_until_visible(driver=driver, xpath=xpath, duration=5)
+            driver.find_element_by_xpath(xpath).click()
+        
+            password_input = driver.find_element_by_xpath("//input[@name='password']")
+            password_input.clear()
+            password_input.send_keys(password)
+
+            LOGGER.info("Logging in")
+            driver.find_element_by_xpath("//input[@value='SIGN IN']").click()
+            sleep_val += random.randint(0,10)
+
+            if num_retries_attempted < num_retries:
+                num_retries_attempted += 1
+                continue
+            else:
+                LOGGER.info("Too many login attempts. Please restart app.")
+                break
+        except Exception as e:
+            LOGGER.exception("Error dialog did not load, proceed. Error: " + str(e))
+            break
 
+    wait_until_visible(driver=driver, xpath="//a[@data-path='myAccount:greeting']")
```

Comment:
Another option is to wait 5s for this to appear, and if it doesn't, catch the error and call `login(...)` again (which would refresh the page and everything). Then you don't have to worry about clicking the error dialog and whatnot. The logic may be a little simpler (and less prone to error) idk. Just throwing out ideas.

For login, speed isn't really a crucial thing since you'd almost always want to login ahead of time anyway using `--login-time`. So full refresh wouldn't be the end of the world.

But at the same time, idk if a full refresh would even work in this case. Would have to test it.

Reply:
I can add that and test, but I have encountered cases where even page refreshes results in that error dialog appearing. What prompted this workaround, on a whim, I just manually clicked that error dialog when it appeared and the error dialog did not appear. This new loop logic, the error dialog did appear once or twice, but eventually allowed login to continue.